distribution/client/src/lib/update.js

```diff
@@ -86,13 +86,19 @@ class Update {
     /*
      * Queue up all the downloads simultaneously, we need updates ASAP!
      */
-    updates.plugins.updates.forEach((plugin) => {
-      logger.info('Downloading', plugin);
-      tasks.push(Downloader.download(plugin.url,
-        Storage.pluginsDirectory(),
-        `${plugin.artifactId}.hpi`,
-        plugin.checksum.signature));
-    });
+    if (updates.plugins.updates) {
+      updates.plugins.updates.forEach((plugin) => {
+        logger.info('Downloading', plugin);
+        tasks.push(Downloader.download(plugin.url,
+          Storage.pluginsDirectory(),
+          `${plugin.artifactId}.hpi`,
+          plugin.checksum.signature));
+      });
+    }
+
+    if (updates.plugins.deletes) {
+      tasks.push(Storage.removePlugins(updates.plugins.deletes));
```

Comment:
Can we add more `logger.info` and the likes inside those `if` conditions? I feel at this stage of the project, we want rather more logs than fewer don't we?